src/librbd/deep_copy/ImageCopyRequest.cc

```diff
@@ -14,7 +14,8 @@
 #define dout_subsys ceph_subsys_rbd
 #undef dout_prefix
 #define dout_prefix *_dout << "librbd::deep_copy::ImageCopyRequest: " \
-                           << this << " " << __func__ << ": "
+                           << this << " " << __func__ << ": src_img_id=" \
+			    << m_src_image_ctx->id << " "
```

Comment:
We don't generally do this in librbd.  Instead, the approach is to include the state machine pointer in all log messages and then to "associate" that pointer with parameters of interest in the state machine constructor (if needed).  I think the only exception is a couple of state machines that format the entire object name instead of printing just the object number.  Otherwise it's hard to know where to stop -- in a pursuit to make each log message stand on its own they tend to get _very_ long. 

So rather than duplicating the same piece(s) of information in each log message, I would suggest to apply what is done e.g. in `ObjectCopyRequest` constructor to other classes.  In `ImageCopyRequest` constructor, you could add something like:
```
  ldout(m_cct, 20) << "src_image_id=" << m_src_image_ctx->id
                   << ", dst_image_id=" << m_dst_image_ctx->id
                   << ", src_snap_id_start=" << m_src_snap_id_start
                   << ", src_snap_id_end=" << m_src_snap_id_end
                   << ", dst_snap_id_start=" << m_dst_snap_id_start
                   ...
```

Reply:
Note that I have spelled out `*_image_id` in full above.  It's not shortened to `img_id` anywhere else in librbd, see `parent_image_id`, `global_image_id`, etc.

Reply:
This doesn't really help for debugging as it still means we eed to go class by class instead of seeing the entire flow of the operation across multiple classes by just searching on an image id or an object id. The above approach would still mean tracking individual class instances which is the problem the PR attempts to solve.

Reply:
> This doesn't really help for debugging as it still means we need to check for individual class instances instead of seeing the entire flow of the operation across multiple classes by just searching on an image id or an object id.

This is understood and more or less intended.  Unless you overhaul the entire librbd such that every log message includes every possible identifier, you wouldn't be able to run a single grep to see the entire flow and discard the rest of the log.  For operations that may involve multiple images, a combination of all those identifiers would need to be included for an arbitrary grep to "just work" -- consider a case where clone image has a long parent chain.  Even here, you included just the source image ID for `ImageCopyRequest`, but someone might be more interested in the destination image ID.

The current approach requires running multiple greps, but as long as you print all desired identifiers in the state machine constructor, the connection between one state machine to the next would be there.  The first grep would produce a list of state machine instances to look at; they can then be extracted in bulk in the second grep.

Reply:
Logs with the updated code:
```

2023-11-03T18:31:59.893+0530 7f6532480540 20 librbd::deep_copy::SnapshotCopyRequest: 0x5609e0a1b880 SnapshotCopyRequest: src_image_id=101f5dda5f6, dst_image_id=10306f6a9b99, src_snap_id_start=0, src_snap_id_end=18446744073709551614, dst_snap_id_start=0, src_snap_ids=4,5,6, dst_snap_ids=
...
2023-11-03T18:31:59.893+0530 7f6532480540 20 librbd::deep_copy::SnapshotCreateRequest: 0x5609e0a1be20 SnapshotCreateRequest: dst_image_id=10306f6a9b99, snap_name=snap1, snap_namespace=[user], size=8388608, parent_spec=[pool_id=-1, pool_namespace=, image_id=, snap_id=head]
...

2023-11-03T18:31:59.893+0530 7f6532480540 20 librbd::deep_copy::SetHeadRequest: 0x5609e0a1bf90 SetHeadRequest: image_id=10306f6a9b99, size=8388608, parent_spec=[pool_id=-1, 
pool_namespace=, image_id=, snap_id=head]
...
2023-11-03T18:32:02.380+0530 7f651effd640 20 librbd::deep_copy::ImageCopyRequest: 0x7f6500015480 ImageCopyRequest: src_image_id=101f5dda5f6, dst_image_id=10306f6a9b99, src_snap_id_start=0, src_snap_id_end=18446744073709551614, dst_snap_id_start=0
...
2023-11-03T18:32:02.383+0530 7f651effd640 20 librbd::deep_copy::ObjectCopyRequest: 0x7f6500021620 ObjectCopyRequest: src_image_id=101f5dda5f6, dst_image_id=10306f6a9b99, dst_oid=rbd_data.10306f6a9b99.0000000000000000, src_snap_id_start=0, dst_snap_id_start=0, snap_map={4=[13],5=[14,13],6=[15,14,13],18446744073709551614=[18446744073709551614,15,14,13]}
...
2023-11-03T18:32:02.473+0530 7f651e7fc640 20 librbd::deep_copy::MetadataCopyRequest: 0x7f64ec018020 MetadataCopyRequest: src_image_id=101f5dda5f6, dst_image_id=10306f6a9b99

```